In [1]:
# Import libraries
import warnings
warnings.filterwarnings('ignore')

import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from mizani.formatters import percent_format
import os
from plotnine import *
import numpy as np
import sys
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf
from stargazer import stargazer
from statsmodels.tools.eval_measures import mse,rmse

In [2]:
#Load original data which from cleaned_data folder on OSF
full_data = pd.read_csv('https://raw.githubusercontent.com/thimaipham/Data/main/morg-2014-emp.csv', 
                        dtype={16: str})

In [4]:
#Check data infomation
full_data.head(2)

Unnamed: 0            hhid intmonth stfips     weight  earnwke  uhours  \
0           3   2600310997690  January     AL  3151.6801   1692.0      40   
1           5  75680310997590  January     AL  3457.1138    450.0      40   

   grade92  race  ethnic  ...  ownchild  chldpres            prcitshp  state  \
0       43     1     NaN  ...         0         0  Native, Born In US     63   
1       41     2     NaN  ...         2         6  Native, Born In US     63   

                            ind02 occ2012                class unionmme  \
0      Employment services (5613)     630  Private, For Profit       No   
1  Outpatient care centers (6214)    5400  Private, For Profit       No   

   unioncov            lfsr94  
0        No  Employed-At Work  
1        No   Employed-Absent  

[2 rows x 23 columns]

In [5]:
# Filter the dataset for the occupation 
# "Lawyers, Judges, magistrates, and other judicial workers" (occupation code 2100)
occupation_code = 2100
data = full_data[full_data['occ2012'] == occupation_code]

# Count the number of observations for this occupation
data.shape[0]

1027

###### This filter data include 1027 observations

### Feture Engineering

In [10]:
# Checking for missing values
missing_values = data.isnull().sum()

# Examining data types
data_types = data.dtypes

# Creating a new column for earnings per hour
data['earnings_per_hour'] = data['earnwke'] / data['uhours']

# Display the results
print(missing_values, data_types) 
print(data[['earnwke', 'uhours', 'earnings_per_hour']].head())


Unnamed: 0             0
hhid                   0
intmonth               0
stfips                 0
weight                 0
earnwke                0
uhours                 0
grade92                0
race                   0
ethnic               981
age                    0
sex                    0
marital                0
ownchild               0
chldpres               0
prcitshp               0
state                  0
ind02                  0
occ2012                0
class                  0
unionmme               0
unioncov              57
lfsr94                 0
earnings_per_hour      0
dtype: int64 Unnamed: 0             int64
hhid                   int64
intmonth              object
stfips                object
weight               float64
earnwke              float64
uhours                 int64
grade92                int64
race                   int64
ethnic               float64
age                    int64
sex                    int64
marital                int64
ownchild  

##### Missing values:
- The `ethnic` column has 981 missing values.
- The `unioncov` column has 57 missing values.
Due to a huge number of missing values in `ethic` column and some missing values in `unioncov` column, I will not choose them in the models of the following steps.  Because we have many other variable options for the models.

##### New column: 
- The column `earnings_per_hour` is created by `earnwke`/`uhours`. This will be the target variable for predictive modeling.